packages/flame/lib/src/components/core/component.dart

```diff
@@ -198,6 +199,11 @@ class Component {
   bool get isLoaded => (_state & _loaded) != 0;
   void _setLoadedBit() => _state |= _loaded;
 
+  @internal
+  bool get isMounting => (_state & _mounting) != 0;
```

Comment:
This is a breaking change if `onGameResize` doesn't fire before `onMount` any more.
If you size your components according to the game size and the game size changes when the component isn't in the tree your component's size will be outdated.
(Added this in here so that it has a discussion thread)

Reply:
Ok, I'll add the Migration Instructions section for this.

Reply:
I've also added the `OnGameResizeBeforeMount` mixin to make the transition easier.

Reply:
What is the reason for not keeping the behavior like it is today?

Reply:
Because we also call `onGameResize` before `onLoad`, and then if we also call it people would start complaining that it's been called too often. So, to avoid that, we need to have an extra status "justFinishedLoadingButHaventStartedMounting", which adds a lot of extra complexity with regards to what to do when this component is removed, or when an actual `onGameResize` is triggered at that time. (We dealt with this problem before by keeping the `isLoading` flag on after the component finished loading, which becomes problematic because if this component is suddenly removed, then it'll get stuck in this fake-loading state forever).

It's a whole lot easier to ditch all that complexity and let the users query the game size manually if they need it (which, I presume, is a tiny fraction of use cases).

Reply:
Feels a bit inconsistent to even call the `onGameResize` before `onLoad` then, it doesn't really serve a purpose there then?
Maybe it could just be called before `onMount` instead, and remove the call before `onLoad`? I think you once suggested that, but we were hesitant to change the lifecycle.

Reply:
Yes, we had a discussion about this in #1421. Ideally, `onGameResize()` would run only when the game actually resizes, as the name suggests. For all other purposes, the canvas size can be obtained manually via `findGame().size` both from `onLoad` and `onMount`. So the best course of action would be to remove `onGameResize` before component loads/mounts altogether.

But that would be a breaking change, so we might not want to do it just yet.

If there is a choice to call either before `onMount` or before `onLoad`, then the first option is easier (because then we don't need to worry about repeating the event if a game resize happens while the component is loading). So, if you're ok with that, I'll change the logic to only call game-resize before `onMount` but not before `onLoad`.

Reply:
> If there is a choice to call either before onMount or before onLoad, then the first option is easier (because then we don't need to worry about repeating the event if a game resize happens while the component is loading). So, if you're ok with that, I'll change the logic to only call game-resize before onMount but not before onLoad.

Yeah, I know I didn't want that previously, but I think it makes more sense to do it that way now and we'll re-think completely the existence and purpose of `onGameResize` for v2.

Reply:
Just making sure I understood the discussion.

`onGameResize` will be called after `onLoad` and before `onMount`, but the game size will be available in both `onLoad` and `onMount` if the user tries to access it right?

Reply:
It would indeed be able to get the game's size in `onLoad` since when changing a parent the first thing that happens is that the parent field is set and this is what `HasGameRef` uses to traverse up to the game where you can get the size.